client/src/components/GraphViewerComponent/GraphViewerRelationshipCreateComponent/GraphViewerRelationshipCreateComponent.js

```diff
@@ -109,40 +122,68 @@ export class GraphViewerRelationshipCreateComponent extends Component {
+      const relationshipItems = await new ModelService().getRelationships(source.modelId, target.modelId);
+      this.setState({ hasRelationships: relationshipItems.length > 0, relationshipItems });
+    } catch (exc) {
+      this.setState({ relationshipItems: [] });
+      exc.customMessage = `Error in retrieving model. Requested ${sourceModelId}`;
+      eventService.publishError(exc);
+    }
+    this.setState({ isLoading: false });
   }
 
   render() {
-    const { relationshipItems, relationshipId, isLoading, showModal, hasRequiredRelError } = this.state;
-    const { sourceId, targetId } = this.getNodes();
+    const { relationshipItems, relationshipId, isLoading, showModal, hasRequiredRelError, sourceId, targetId, hasRelationships } = this.state;
 
     return (
       <ModalComponent isVisible={showModal} isLoading={isLoading} className="gc-dialog">
         <h2 className="heading-2" tabIndex="0" id="create-relationship-heading">Create Relationship</h2>
         <h4>Source ID</h4>
         <TextField disabled readOnly id="sourceIdField" ariaLabel="Source ID" className="modal-input" styles={this.getStyles} value={sourceId} />
+        <div className="btn-icon">
+          <IconButton iconOnly="true" iconProps={this.swapRelationshipIcon} title="Swap Relationship" ariaLabel="Swap Relationship" onClick={this.swap} />
+        </div>
         <h4>Target ID</h4>
         <TextField disabled readOnly id="targetIdField" ariaLabel="Target ID" className="modal-input" styles={this.getStyles} value={targetId} />
-        <h4>Relationship</h4>
-        <Dropdown
-          tabIndex="0"
-          ariaLabel="Select an option"
-          required
-          placeholder="Select an option"
-          className="modal-input"
-          selectedKey={relationshipId}
-          options={relationshipItems.map((q, i) => ({ key: i, text: q }))}
-          styles={{
-            dropdown: { width: 208 }
-          }}
-          errorMessage={hasRequiredRelError ? "Please select a relationship" : null}
-          onChange={this.onSelectedRelChange} />
+        <h4>Relationship</h4> {
+          hasRelationships && <div>
+            <Dropdown
+              tabIndex="0"
+              ariaLabel="Select an option"
+              required
+              placeholder="Select an option"
+              className="modal-input"
+              selectedKey={relationshipId}
+              options={relationshipItems.map((q, i) => ({ key: i, text: q }))}
+              styles={{
+                dropdown: { width: 208 }
+              }}
+              errorMessage={hasRequiredRelError ? "Please select a relationship" : null}
+              onChange={this.onSelectedRelChange} />
+          </div>
+        } {
+          !hasRelationships && <div className="warning-icon-wrapper">
+            <div className="warning-icon">
+              <IconButton iconOnly="true" iconProps={this.warningRelationshipIcon} title="Warning Relationship" ariaLabel="Warning  Relationship" />
```

Comment:
Let's get rid of the warningRelationshipIcon property on the class.

Instead we can just specify it like `iconProps={{ iconName: xxx }}`

Reply:
Ready